pkg/broker/ttl.go

```diff
@@ -48,3 +52,47 @@ func SetTTL(ctx cloudevents.EventContext, ttl int32) error {
 func DeleteTTL(ctx cloudevents.EventContext) error {
 	return ctx.SetExtension(TTLAttribute, nil)
 }
+
+// TTLDefaulter returns a cloudevents event defaulter that will manage the TTL
+// for events with the following rules:
+//   If TTL is not found, it will set it to the default passed in.
+//   If TTL is <= 0, it will remain 0.
+//   If TTL is > 1, it will be reduced by one.
+func TTLDefaulter(logger *zap.Logger, defaultTTL int32) client.EventDefaulter {
+	return func(ctx context.Context, event cloudevents.Event) cloudevents.Event {
+		// Get the current or default TTL from the event.
+		var ttl int32
+		if ttlraw, err := event.Context.GetExtension(TTLAttribute); err != nil {
+			logger.Debug("TTL not found in outbound event, defaulting.",
+				zap.String("event.id", event.ID()),
+				zap.Int32(TTLAttribute, defaultTTL),
+				zap.Error(err),
+			)
+			ttl = defaultTTL + 1
```

Comment:
why the +1? here and below? If it's correct, can you add a comment, because it might be confusing for a future mantainer

Reply:
if default ttl was 1, then the message would never go. we dec ttl on line 82, the old function had this bug

Reply:
I was able to move in the subtraction in an else

Reply:
thanks!